e2e/explorer/workflow/revisions/trafficShaping.spec.ts

```diff
@@ -0,0 +1,162 @@
+import { createNamespace, deleteNamespace } from "../../../utils/namespace";
+import { expect, test } from "@playwright/test";
+
+import { noop as basicWorkflow } from "~/pages/namespace/Explorer/Tree/NewWorkflow/templates";
+import { createWorkflow } from "~/api/tree/mutate/createWorkflow";
+import { createWorkflowWithThreeRevisions } from "../../../utils/revisions";
+import { faker } from "@faker-js/faker";
+
+let namespace = "";
+
+test.beforeEach(async () => {
+  namespace = await createNamespace();
+});
+
+test.afterEach(async () => {
+  await deleteNamespace(namespace);
+  namespace = "";
+});
+
+test("by default, traffic shaping is not enabled", async ({ page }) => {
+  const name = faker.system.commonFileName("yaml");
+
+  await createWorkflow({
+    payload: basicWorkflow.data,
+    urlParams: {
+      baseUrl: process.env.VITE_DEV_API_DOMAIN,
+      namespace,
+      name,
+    },
+  });
+
+  await page.goto(`/${namespace}/explorer/workflow/revisions/${name}`);
+  await expect(
+    page.getByTestId("traffic-shaping-container"),
+    "it renders the traffic shaping component"
+  ).toBeVisible();
+
```

Comment:
Some general thoughts where I'm interested in your feedback:

In some places I use hard coded texts from the translation files and I wonder if we can keep it that way. Generally I would prefer to use a test id, but I have some cases where it seems to make sense. The downside is, that the tests break when we change some translations, but it's also a very easy fix.

Examples where I use text in this file:

"Select Revision..." this is the "best" way to find out if the dropdown is unused. I could add a helper attribute to the button, but then I would just test the data attribute.

"Please select 2 different revisions to configure traffic shaping." - I would prefer this, instead of introducing a redundant DOM element

"The traffic will be split between NAMEA and NAMEB with a ratio of X to Y %" - I want to be very sure that the user sees this text and also that we fill the placeholder from the translation file correctly